src/Uno.UI/UI/Xaml/ResourceDictionary.cs

```diff
@@ -48,7 +72,15 @@ public Uri Source
 		}
 
 		public IList<ResourceDictionary> MergedDictionaries => _mergedDictionaries;
-		public IDictionary<object, object> ThemeDictionaries => _themeDictionaries ??= new ResourceDictionary();
+		public IDictionary<object, object> ThemeDictionaries
+		{
+			get
+			{
+				_themeDictionaries ??= new ResourceDictionary();
+				RegisterOnKeyAdded();
```

Comment:
Registering on every get is probably expensive :)

I'd rather see an observable collection and a "version" integer or something similar to determine that the merged dictionary has changed. Also, can you make a different PR for this, specifically? Thanks!

Reply:
@jeromelaban My intention was to register only when it's re-assigned (ie, if it was null), but I forgot the `if` condition 😄 
Would that work or you think I should go for the an observable collection anyway?

Reply:
No, in retrospect it would not be enough. I'm wondering about the performance impact, though it'll affect setting values, not reading, so it's less of a problem.